packages/xo-server/src/xo-mixins/metadata-backups.js

```diff
@@ -0,0 +1,261 @@
+// @flow
+import asyncMap from '@xen-orchestra/async-map'
+import defer from 'golike-defer'
+import { finished } from 'readable-stream'
+import { promisify, ignoreErrors } from 'promise-toolbox'
+
+import { type Xapi } from '../xapi'
+import {
+  safeDateFormat,
+  type SimpleIdPattern,
+  unboxIdsFromPattern,
+} from '../utils'
+
+import { type Executor, type Job } from './jobs'
+import { type Schedule } from './scheduling'
+
+const pFinished = promisify(finished)
+
+const METADATA_BACKUP_JOB_TYPE = 'metadataBackup'
+
+type Settings = {|
+  retentionXoMetadata?: number,
+  retentionPoolMetadata?: number,
+|}
+
+type MetadataBackupJob = {
+  ...$Exact<Job>,
+  pools?: SimpleIdPattern,
+  remotes: SimpleIdPattern,
+  settings: $Dict<Settings>,
+  type: METADATA_BACKUP_JOB_TYPE,
+  xoMetadata?: boolean,
+}
+
+const METADATA_BACKUP_DIR = 'xo-metadata-backups'
+
+// File structure on remotes:
+//
+// <remote>
+// └─ xo-metadata-backups
```

Comment:
I think we should split pool metadata and xo configs in separate directories.

Reply:
It's for separation of concerns or it can cause troubles?

Reply:
Simply separation.

So that we can save one directory or another.

Reply:
```
// File structure on remotes:
//
// <remote>
// ├─ xo-metadata
// │  └─ <schedule ID>
// │     └─ <YYYYMMDD>T<HHmmss>
// │        ├─ metadata.json
// │        └─ data.json
// └─ pool-metadata
//    └─ <schedule ID>
//       └─ <YYYYMMDD>T<HHmmss>
//          ├─ metadata.json
//          └─ data
```

Reply:
For consistency with `xo-vm-backups` I would suggest:

- `xo-config-backups`
- `xo-pool-metadata-backups`

@pdonias What do you think?

Reply:
Yes!